subsys/net/lib/sockets/sockets.c

```diff
@@ -577,3 +591,49 @@ int zsock_inet_pton(sa_family_t family, const char *src, void *dst)
 		return 0;
 	}
 }
+
+int zsock_getsockopt(int sock, int level, int optname,
+		     void *optval, socklen_t *optlen)
+{
+	int ret = 0;
+
+	if (!sock)
+		SET_ERRNO(-ENOTSOCK);
+
+	switch (level) {
+	case SOL_TLS:
+		ret = tls_getsockopt(sock, level, optname, optval, optlen);
+		break;
+
+	default:
+		ret = -EOPNOTSUPP;
+		break;
+	}
+
+	SET_ERRNO(ret);
+
+	return 0;
+}
+
+int zsock_setsockopt(int sock, int level, int optname,
```

Comment:
Where will the setsockopt() options be passed to the offload driver?
Is the plan to add setsockopt()/getsockopt() functions to the include/net_offload.h:struct net_offload, or will we be able to do socket offload in the future?

Reply:
That's not yet implemented... somewhere identical to other socket offloading support, I'd say. This PR starts to be quite large, maybe offloading goes into a PR of its own once this initial support is merged.

Reply:
Issue #8277 created to track this.